app/controllers/api/stages_controller.rb

```diff
@@ -19,10 +18,25 @@ def clone
     end
   end
 
+  def update
+    stage.deploy_groups = deploy_groups
+    stage.save!
+    head :no_content
+  end
+
+  def template_stage
+    render json: current_project.template
```

Comment:
404 will raise in an api client, so we don't necessarily want to do that... right?